core/src/main/java/org/bitcoinj/evolution/QuorumState.java

```diff
@@ -111,12 +140,25 @@ public void applyDiff(Peer peer, AbstractBlockChain headersChain, AbstractBlockC
             quorumList.syncWithMasternodeList(newMNList);
         }
         if (peer != null && isSyncingHeadersFirst) peer.queueMasternodeListDownloadedListeners(MasternodeListDownloadedListener.Stage.ProcessedQuorums, mnlistdiff);
-        mnListsCache.put(newMNList.getBlockHash(), newMNList);
         quorumsCache.put(newQuorumList.getBlockHash(), newQuorumList);
-        mnList = newMNList;
         quorumList = newQuorumList;
     }
 
+    @Override
+    int getBlockHeightOffset() {
+        return SigningManager.SIGN_HEIGHT_OFFSET;
+    }
+
+    @Override
+    public int getUpdateInterval() {
+        return 1;
+    }
```

Comment:
With DIP4, every time that a block is added to the blockchain, we will request an update for the masternode/quorum lists.